### ISSUE TITLE
Unable to deploy app using docker

### ISSUE THREAD

**Issue by masonedmison** [gvolpe/pfps-shopping-cart#448]
When trying to deploy the app using the compose file in the app/ directory as is, the app port doesn't appear to be exposed to the host (unable to hit the endpoints using the imsomnia.json import to insomnia and `lsof -i tcp:8080` yield empty result). I tried adding the "ports"  option with a value of "8080:8080" in the compose file but the "ports" option is not compatible with "network_mode". When I remove network mode and include the "ports" option, I am then unable to connect to the databases and get a "ConnectionRefusedError".  It looks like the port should be exposed by the "DockerPlugin" but it doesn't seem to work for me. Can you please advise?

The logs show http4s starting at 
```
HTTP Server started at /0:0:0:0:0:0:0:0:8080
```
which looks weird. I have 
```
        HttpServerConfig(
          host = host"0.0.0.0",
          port = port"8080"
```
in `Config`

Very, very helpful and awesome book by the way.  A little over halfway through and have already learned so very much :)

**Comment by masonedmison** [comment]
Poking around a little more, it looks like the network_mode setting only works on Linux (I'm on mac so this seems to be the issue [seen here](https://docs.docker.com/network/host/). Now removing `network_mode` and adding `ports: -8080:8080` I get 
```
shopping_cart_1  | 01:27:15.052 [io-compute-3] INFO  shop.Main - Releasing Redis connection: RedisURI(redis://localhost)
shopping_cart_1  | Exception in thread "main" java.net.ConnectException: Connection refused
shopping_cart_1  | 	at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.checkConnect(Native Method)
shopping_cart_1  | 	at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.finishConnect(Unknown Source)
shopping_cart_1  | 	at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.finish(Unknown Source)
shopping_cart_1  | 	at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.onEvent(Unknown Source)
shopping_cart_1  | 	at java.base/sun.nio.ch.EPollPort$EventHandlerTask.run(Unknown Source)
shopping_cart_1  | 	at java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(Unknown Source)
shopping_cart_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
shopping_cart_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
shopping_cart_1  | 	at java.base/java.lang.Thread.run(Unknown Source)
```

Looks like it's unable to connect to Redis. I am spinning up the Redis instance and Postgres db instances using the outermost `docker-compose.yml` file. Both look to be running so I am not sure why I am getting this connection error.

**Comment by gvolpe** [comment]
I don't use Mac so I can't help you much. You'll need to figure out how to get Docker working on Mac.

I would suggest asking in the Gitter channel, I know plenty of readers use Mac and have Docker working without issues.

**Comment by hagl** [comment]
I had the same issues on MacOS. The postgres and redis instances are mapped to the host via the outer docker-compose.yml file, but are not accessible from the container inside, since network_mode: host is not supported on mac os.

To be able to run the app I used the following approach:
I put all three services (postgres, redis and shopping_cart) in one docker-compose file and made `shopping_cart` depend_on the other two. In that case redis is accessible under redis:6739 and postgres under postgres:5432. To handle these different URLs, I created a new AppEnvironement `Docker` and pass the postgresHost in the Config.default method. See attached patch:

[docker-mac-os.patch.gz](https://github.com/gvolpe/pfps-shopping-cart/files/6926226/docker-mac-os.patch.gz)
